Wip rf entry

ronen-fr opened the pull request:
NOT READY

athanatos reviewed (approved):
Looks very reasonable!  I really appreciate the pointers in the commit messages --made for a very smooth review.

ronen-fr commented:
Added unit-tests for the additional functionality of the not-before-queue;
Fixed all issues detected by the Teuthology runs.
My plan is to rerun the Teu tests, and then to merge.

NitzanMordhai reviewed (approved):
the added standalone commits look good!
